src/Analyzer/NameResolver.php

```diff
@@ -341,4 +388,52 @@ private function resolveType($node)
 
         return $node;
     }
+
+    private function getPhpDocNode(NodeAbstract $node): ?PhpDocNode
+    {
+        if (null === $node->getDocComment()) {
+            return null;
+        }
+
+        /** @var Doc $docComment */
+        $docComment = $node->getDocComment();
+
+        $tokens = $this->phpDocLexer->tokenize($docComment->getText());
+        $tokenIterator = new TokenIterator($tokens);
+
+        return $this->phpDocParser->parse($tokenIterator);
+    }
+
+    /**
+     * @param Node\Param|Stmt\Property $node
+     */
+    private function isNodeOfTypeArray($node): bool
+    {
+        return null !== $node->type && isset($node->type->name) && 'array' === $node->type->name;
+    }
+
+    private function getArrayItemType(TypeNode $typeNode): ?string
+    {
+        $arrayItemType = null;
+
+        if ($typeNode instanceof GenericTypeNode) {
+            if (1 === \count($typeNode->genericTypes)) {
+                $arrayItemType = (string) $typeNode->genericTypes[0];
+            } elseif (2 === \count($typeNode->genericTypes)) {
+                $arrayItemType = (string) $typeNode->genericTypes[1];
```

Comment:
Why do you take the second parameter if the count is 2?

Reply:
When count is 2 we are handling the case `array<int, MyDto>` in which `$typeNode->genericTypes[0]` is the key of the array (in this case `int`) and `$typeNode->genericTypes[1]` is the the type of the item (in this case `MyDto`). This code makes sense since we already checked that the current node is an `array`.

Reply:
Some extra information.
When count is 1, we are handling case `list<MyDto>` (and also `array<MyDto>` but this is not a 100% correct syntax).
When `$typeNode instanceof ArrayTypeNode` we are handling the case `MyDto[]` which is called "legacy syntax" since with PHPStan and PSalm is not longer used

Reply:
Ok, now is more clear, I don't know if there is a way to explicit it in the code without a lot of comments.

Reply:
The 3 cases are explicit in the test cases. If you prefer I can add a short comment for each case.

Reply:
For me a small comment can work fine, I was thinking to change something in the code to be more explicit like a function with a specific name instead of directly the `if` condition

Reply:
Using a function will be something like that:
```
if ($this->isTypedArrayNodeWithListSyntax($typeNode)) {
    $arrayItemType = (string) $typeNode->genericTypes[0];
}
```

but the knowledge about `$typeNode->genericTypes` will be splitted in many points increasing code readability. In this case I think that a small comment is better.

Reply:
A small comment is better I think, good enough at the moment @LuigiCardamone